src/Microsoft.ML.Data/Transforms/ColumnSelecting.cs

```diff
@@ -725,6 +727,22 @@ DataViewRow IRowToRowMapper.GetRow(DataViewRow input, IEnumerable<DataViewSchema
 
             IDataTransform ITransformTemplate.ApplyToData(IHostEnvironment env, IDataView newSource)
                 => new SelectColumnsDataTransform(env, _transform, new Mapper(_transform, newSource.Schema), newSource);
+
+            public bool CanSaveOnnx(OnnxContext ctx) => true;
+
+            public void SaveAsOnnx(OnnxContext ctx)
+            {
+                var outputToInputMap = _mapper.OutputToInputMap;
+                for(int i = 0; i < outputToInputMap.Length; i++)
+                {
+                    var srcCol = InputSchema[outputToInputMap[i]];
+                    var dstCol = OutputSchema[i];
+                    var srcVariable = ctx.GetVariableName(srcCol.Name);
+                    var dstVariable = ctx.AddIntermediateVariable(dstCol.Type, dstCol.Name, true);
+                    string opType = "Identity";
+                    ctx.CreateNode(opType, srcVariable, dstVariable, ctx.GetNodeName(opType), "");
+                }
```

Comment:
Where are you dropping remaining columns?

Reply:
By default, only the columns that have connections are selected. So the columns not in the OutputToInputMap are automatically dropped.

But your comment made me look deeper into the output schema. I have fixed the output schema with an updated to OnnxTransformer.

---
In reply to: [361511119](https://github.com/dotnet/machinelearning/pull/4590#discussion_r361511119) [](ancestors = 361511119)